Common/src/containers/CFileReaderLUT.cpp

```diff
@@ -231,22 +233,62 @@ void CFileReaderLUT::ReadRawDRG(const string& file_name) {
   if (n_hull_points != hullCounter - 1)
     SU2_MPI::Error(
         "Number of read hull points does not match number of points "
-        "specified in Dragon library header.",
+        "specified in lookup table library header.",
         CURRENT_FUNCTION);
 
   file_stream.close();
 
 }
 
-string CFileReaderLUT::SkipToFlag(ifstream* file_stream, const string& flag) {
-  string line;
-  getline(*file_stream, line);
 
-  while (line.compare(flag) != 0 && !(*file_stream).eof()) {
-    getline(*file_stream, line);
+ /*! \brief Searches for the position of flag in file_stream and 
+  *         sets the stream position of file_stream to that position.
+  */
+void CFileReaderLUT::SkipToFlag(ifstream& file_stream, const string& current_line, const string& flag) const {
+  string next_line = "";
+
+  /*--- compare current line to flag and return if equal ---*/
+  if (current_line.compare(flag) == 0) return;
+
+  /*--- else, search for flag ---*/
+  while (next_line.find(flag) == string::npos && !(file_stream).eof()) {
+    GetStrippedLine(file_stream, next_line);
   }
 
-  if ((*file_stream).eof()) SU2_MPI::Error("Flag not found in file", CURRENT_FUNCTION);
+  /*--- throw error if end of file reached ---*/
+  if ((file_stream).eof()) SU2_MPI::Error("Flag " + flag + " not found in file", CURRENT_FUNCTION);
+}
 
-  return line;
+/*! \brief Extracts the next non-empty characters from file_stream and stores them into line.
+ */
+bool CFileReaderLUT::GetNextNonEmptyLine(ifstream& file_stream, string& line) const {
+
+  /*--- get next line and save return value ---*/
+  bool return_value = GetStrippedLine(file_stream, line);
+  
+  /*--- skip empty lines ---*/
+  while (line.empty() && !(file_stream).eof()){
+    return_value = GetStrippedLine(file_stream, line);
+  }
+
+  /*--- return true if line is not empty, else return false ---*/
+  return return_value;
 }
+
+/*! \brief Extracts characters from file_stream, removes trailing control characters,
+ *         and stores them into line.
+ */
```

Comment:
Btw it is usually better to put the doxygen comment on the declaration hpp, but it's fine

Reply:
Thanks, I've moved it to the hpp files. When it goes through the regression tests, I'll merge.